File: model/ChartView.go
package model

import (
	"encoding/json"
	"fmt"
)

type ChartView struct {
	ViewBase
	ThemBase
	EventBase
	HeaderHeight float32  `json:"headerHeight"`
	ChartHeight  float32  `json:"chartHeight"`
	FooterHeight float32  `json:"footerHeight"`
	Avatar       string   `json:"avatar"`
	Title        string   `json:"title"`
	Caption      string   `json:"caption"`
	DataArray    []int32  `json:"dataArray"`
	RowArray     []string `json:"rowArray"`
	ColumnArray  []string `json:"columnArray"`
}

var chartViews []ChartView

func AddChartView(jsonString string) {
	var view ChartView

	err := json.Unmarshal([]byte(jsonString), &view)
	if err != nil {
		fmt.Println("Error AddChartView: unmarshalling JSON:", err)
	} else {
		chartViews = append(chartViews, view)
	}
}

File: web/packages/base/ChartView.ts
import type {ViewProps, ViewValues} from "../view";
import {ThemValues, TouchProps} from "../view";

export interface ChartViewProps extends ViewProps, TouchProps {

    backgroundColor?: string,
    activeColor?: string,

    headerHeight?: number,
    chartHeight?: number,
    footerHeight?: number,

    avatar: string
    title: string
    caption?: string

    dataArray: number[],
    rowArray: string[],
    columnArray: string[],
}

export type ChartViewValues = ViewValues & ThemValues & {
    backgroundColor?: string,
    activeColor?: string,
    circleColor?: string,

    headerHeight?: number,
    chartHeight?: number,
    footerHeight?: number,

    title: string
    caption?: string

    rowCount: number,
    columnCount: number,
};

export default function ChartView(props: ChartViewProps) {

}


